src/nft/components/details/detailsV2/TabbedComponent.tsx

```diff
@@ -0,0 +1,62 @@
+import Row from 'components/Row'
+import { useState } from 'react'
+import styled from 'styled-components/macro'
+import { ThemedText } from 'theme'
+
+import { containerStyles } from './shared'
+
+const TabbedComponentContainer = styled.div`
+  height: 288px;
+
+  ${containerStyles}
+`
+
+const TabsRow = styled(Row)`
+  gap: 32px;
+  margin-bottom: 12px;
+  width: 100;
+`
+
+const Tab = styled(ThemedText.SubHeader)<{ isActive: boolean; numTabs: number }>`
+  color: ${({ theme, isActive }) => (isActive ? theme.textPrimary : theme.textTertiary)};
+  line-height: 24px;
+  cursor: ${({ numTabs }) => (numTabs > 1 ? 'pointer' : 'default')};
+
+  &:hover {
+    opacity: ${({ numTabs, theme }) => numTabs > 1 && theme.opacity.hover}};
+  }
+`
+
+export interface Tab {
+  title: React.ReactNode
+  key: string
+  content: JSX.Element
+}
+
+interface TabbedComponentProps {
+  tabs: Tab[]
+  defaultTab: Tab
+  style?: React.CSSProperties
+}
+
+export const TabbedComponent = ({ tabs, defaultTab, style }: TabbedComponentProps) => {
+  const [activeTab, setActiveTab] = useState(defaultTab.key)
+  const activeContent = tabs.find((tab) => tab.key === activeTab)?.content
+  return (
+    <TabbedComponentContainer style={style}>
+      <TabsRow>
+        {tabs.map((tab) => (
+          <Tab
+            isActive={activeTab === tab.key}
+            numTabs={tabs.length}
+            onClick={() => setActiveTab(tab.key)}
```

Comment:
so typically the .map pattern you pass the `setActiveTab` function and then a key to the child, and then you don't create new functions in each render. in this case because the tab is a styled component that would require a new wrapper, i don't think it's worth it - but just something to keep in mind.